cmd/kubectl-gadget/utils/trace.go

```diff
@@ -208,77 +198,230 @@ func GenericTraceCommand(
 			Do(context.TODO()).
 			Error()
 		if err != nil {
-			deleteTraces(nil, traceRestClient, traceID)
-			contextLogger.Fatalf("Error creating trace on node %s: %q", node.Name, err)
+			traceID, present := trace.ObjectMeta.Labels[GLOBAL_TRACE_ID]
+			if present {
+				// Clean before exiting!
+				deleteTraces(traceRestClient, traceID)
+			}
+
+			return fmt.Errorf("Error creating trace on node %q: %w", node.Name, err)
 		}
 	}
 
-	if params.Node != "" && !nodeFound {
-		contextLogger.Fatalf("Invalid filter: Node %q does not exist", params.Node)
+	// It is possible we change this value in the above loop, restore it to its
+	// previous value.
+	trace.Spec.Node = traceNode
+
+	if trace.Spec.Node != "" && !nodeFound {
```

Comment:
I think you can directly use `traceNode` in the `if` statement. Or what did I miss?